testplan/report/testing/base.py

```diff
@@ -450,6 +459,12 @@ def merge(self, report, strict=True):
         super(TestGroupReport, self).merge(report, strict=strict)
         self.propagate_tag_indices()
 
+    @property
+    def attachments(self):
+        """Return all attachments from child reports."""
+        return itertools.chain.from_iterable(
```

Comment:
> itertools.chain.from_iterable 

is this returning a array[arrays[attachements]] for  all the TestCaseReports below it?
Why did you choose to implement it with itertools? :D

Reply:
This code is flattening the attachment lists from each child report object into one single list. So for example if our report object has 3 children that each have 3 attachments, it will return a single list containing all 9 attachment objects.

Reply:
I used itertools.chain just because it's a part of the standard library that can be used to flatten multiple iterables into one: https://docs.python.org/3.7/library/itertools.html#itertools.chain.from_iterable